tests/v2/validation/rbac/rbac_test.go

```diff
@@ -0,0 +1,249 @@
+		assert.Equal(rb.T(), namespaceName, createdNamespace.Name)
+		actualStatus := fmt.Sprintf("%v", createdNamespace.Status.Phase)
+		assert.Equal(rb.T(), "Active", actualStatus)
+	case roleMember:
+		require.Error(rb.T(), err)
+		//assert cluster member gets an error when creating a namespace in a project they are not owner of
+		errMessage := strings.Split(err.Error(), ":")[0]
+		assert.Equal(rb.T(), "namespaces is forbidden", errMessage)
+	}
+
+	//Testcase5 Validate if cluster members are able to list all the namespaces in a cluster
+	log.Info("Testcase5 - Validating if ", role, " can lists all namespaces in a cluster.")
+	//Get the list of namespaces as and admin client
+	namespaceListAdmin, err := getNamespaces(rb.client, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	//Get the list of namespaces as an admin client
+	namespaceListClusterMembers, err := getNamespaces(rb.standardUserClient, rb.cluster.ID)
+
+	switch role {
+	case roleOwner:
+		require.NoError(rb.T(), err)
+		//Length of namespace list for admin and cluster owner should match
+		assert.Equal(rb.T(), len(namespaceListAdmin), len(namespaceListClusterMembers))
+		//Namespaces obtained as admin and cluster owner should be same
+		assert.Equal(rb.T(), namespaceListAdmin, namespaceListClusterMembers)
+	case roleMember:
+		require.Error(rb.T(), err)
+		errMessage := strings.Split(err.Error(), ":")[0]
+		//assert cluster member is not able to list namespaces
+		assert.Equal(rb.T(), "namespaces is forbidden", errMessage)
+	}
+
+	//Testcase6 Validate if cluster members are able to delete the namespace in the project they are not owner of
+	log.Info("Testcase6 - Validating if ", role, " can delete a namespace from a project they are not owner of.")
+	err = namespaces.DeleteNamespace(rb.standardUserClient, namespaceName, rb.cluster.ID)
+	switch role {
+	case roleOwner:
+		require.NoError(rb.T(), err)
+	case roleMember:
+		require.Error(rb.T(), err)
+	}
+}
+
+func (rb *rbTestSuite) ValidateDeleteProject(role string) {
+
+	//Testcase7 Validate if cluster members are able to delete the project they are not owner of
+	err := rb.standardUserClient.Management.Project.Delete(rb.adminProject)
+
+	switch role {
+	case roleOwner:
+		require.NoError(rb.T(), err)
+	case roleMember:
+		require.Error(rb.T(), err)
+		errStatus := strings.Split(err.Error(), ".")[1]
+		rgx := regexp.MustCompile(`\[(.*?)\]`)
+		errorMsg := rgx.FindStringSubmatch(errStatus)
+		assert.Equal(rb.T(), "403 Forbidden", errorMsg[1])
+	}
+}
+
+func (rb *rbTestSuite) ValidateRemoveClusterRoles(role string) {
+
+	//Testcase8 Remove added cluster member from the cluster as an admin
+	err := users.RemoveClusterRoleFromUser(rb.client, rb.standardUser)
+	require.NoError(rb.T(), err)
+
+}
+
+func (rb *rbTestSuite) TestRBAC() {
+	tests := []struct {
+		name        string
+		clusterRole string
+	}{
+		{"Cluster Owner", "cluster-owner"},
+		{"Cluster Member", "cluster-member"},
+	}
+	for _, tt := range tests {
+		rb.Run(tt.name, func() {
```

Comment:
We don't need this block of code to be within a Run(). But if you want to keep it, can you please make it more descriptive? In Jenkins it just looks like `Cluster_Owner#1` etc. May something like "`Set Up User With Cluster Role ..." `That way know what aspect of the test failed on first glance